### PR TITLE
non-ephemeral teambot keys

### PR DESCRIPTION
**Pull request by joshblum**
depends on https://github.com/keybase/client/pull/18318, https://github.com/keybase/client/pull/18290, https://github.com/keybase/client/pull/18403, https://github.com/keybase/keybase/pull/4054

- adds new `teambot` package to handle non-ephemeral teambot keys and some shared logic with ephemeral keys
- adds a `BotKeyer` which handles the bot side of key management, storing/unboxing keys made by members and requesting keys when needed
- adds a `MemberKeyer` which derives the `TeambotKey` from a `TeamApplicationKey`, only the `CHAT` application is supported to mirror the ephemeral setup but could be revisited

cc @keybase/hotpotatosquad

### CODE REVIEW

**Comment by mmaxim** [comment]
Is this ready? Still marked draft.

**Comment by joshblum** [comment]
Not yet, going to cleanup after everything is merged down

File: go/ephemeral/teambot_ek.go [hunk 1]
```diff
@@ -359,51 +343,3 @@ func (*TeambotEphemeralKeyer) Fetch(mctx libkb.MetaContext, teamID keybase1.Team
 	}
 	return keybase1.NewTeamEphemeralKeyBoxedWithTeambot(teambotEKBoxed), nil
 }
-
-func notifyTeambotEKNeeded(mctx libkb.MetaContext, teamID keybase1.TeamID, generation keybase1.EkGeneration) (err error) {
```
Comment (joshblum) [inline go/ephemeral/teambot_ek.go]: this moved to the `teambot` pkg

**Comment by joshblum** [comment]
may add background keygen during team rotate, pending discussion

File: go/chat/storage/blockengine_memcache.go [hunk 2]
```diff
@@ -35,15 +35,14 @@ func (c *logContext) GetVDebugLog() *libkb.VDebugLog {
 type blockEngineMemCacheImpl struct {
 	blockCache *lru.Cache
 	logContext *logContext
-	lockTab    *libkb.LockTable
+	lockTab    libkb.LockTable
```
Comment (mmaxim) [inline go/chat/storage/blockengine_memcache.go]: What is this for?

File: go/libkb/notify_router.go [hunk 3]
```diff
@@ -2108,6 +2117,73 @@ func (n *NotifyRouter) HandleTeambotEKNeeded(ctx context.Context, teamID keybase
 	n.G().Log.CDebugf(ctx, "- Sent TeambotEKNeeded notification")
 }
 
+func (n *NotifyRouter) HandleNewTeambotKey(ctx context.Context, teamID keybase1.TeamID,
```
Comment (mmaxim) [inline go/libkb/notify_router.go]: Why do these exist?

Comment (joshblum) [reply go/libkb/notify_router.go]: just used by tests

File: go/chat/storage/blockengine_memcache.go (hunk 2 above)
Comment (joshblum) [reply go/chat/storage/blockengine_memcache.go]: just cleaning up how the other packages use this

Comment (mmaxim) [reply go/chat/storage/blockengine_memcache.go]: How is this even correct? All of `LockTable`'s methods have pointer receivers

**Pull request merged**